ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -248,7 +248,9 @@ public SchemaKStream selectKey(final Field newKeyField) {
               .toString();
       return newKey;
     }).mapValues((key, row) -> {
-      row.getColumns().set(SchemaUtil.ROWKEY_NAME_INDEX, key);
+      if (updateRowKey) {
```

Comment:
we can probably just change this to use `map` instead of `selectKey(..).mapValues()`, i.e,
```
map((key, value) -> {
   // key selection code from above
  // updateRowKey from here
 return KeyValue.pair(newKey, value)
}
```

Reply:
done.

Reply:
I had to back this suggestion out since it was causing the `JoinNodeTest` to fail with an NPE for some reason.